### PR TITLE
Add Flickr postprocessing transform for phrase grounding

### PR DESCRIPTION
**Pull request by ebsmothers**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110
* __->__ #109

Test plan:
Added a unit test under examples/mdetr
python -m pytest examples/mdetr/test/test_transforms.py
======================================= test session starts ========================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /data/home/ebs/torchmultimodal
collected 4 items

examples/mdetr/test/test_transforms.py .... [100%]

======================================== 4 passed in 3.56s =========================================

Differential Revision: [D37390043](https://our.internmc.facebook.com/intern/diff/D37390043)

### CODE REVIEW

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Comment by ebsmothers** [comment]
test

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Comment by ebsmothers** [comment]
test

**Comment by ebsmothers** [comment]
test

**Comment by ebsmothers** [comment]
test

**Comment by ebsmothers** [comment]
test

**Review by ankitade — APPROVED** [review]
stamp

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

File: examples/mdetr/test/test_transforms.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
[the remark this answers is not in the document]
Comment (ebsmothers) [reply examples/mdetr/test/test_transforms.py:1]: Holdover from a previous version, lemme rename

File: examples/mdetr/data/postprocessors.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    def __call__(
+        self, outputs, target_sizes, positive_map, items_per_batch_element
+    ) -> List[Any]:
+        """Perform the computation.
+        Args:
+            outputs: raw outputs of the model
+            target_sizes: tensor of dimension [batch_size x 2] containing the size of each images of the batch
+                          For evaluation, this must be the original image size (before any data augmentation)
+                          For visualization, this should be the image size after data augment, but before padding
+            positive_map: tensor [total_nbr_phrases x max_seq_len] for each phrase in the batch, contains a binary
```
[the remark this answers is not in the document]
Comment (ebsmothers) [reply examples/mdetr/data/postprocessors.py:31]: number. Lemme rename to num to make it clearer

File: examples/mdetr/data/postprocessors.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    def __call__(
+        self, outputs, target_sizes, positive_map, items_per_batch_element
+    ) -> List[Any]:
+        """Perform the computation.
+        Args:
+            outputs: raw outputs of the model
+            target_sizes: tensor of dimension [batch_size x 2] containing the size of each images of the batch
+                          For evaluation, this must be the original image size (before any data augmentation)
+                          For visualization, this should be the image size after data augment, but before padding
```
[the remark this answers is not in the document]
Comment (ebsmothers) [reply examples/mdetr/data/postprocessors.py:30]: Nonexistent for now so will remove

File: examples/mdetr/data/postprocessors.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+from typing import Any, List
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torchvision.ops.boxes import box_convert
+
+
+class PostProcessFlickr:
+    """This module converts the model's output for Flickr30k entities evaluation.
```
[the remark this answers is not in the document]
Comment (ebsmothers) [reply examples/mdetr/data/postprocessors.py:16]: Oh this is just the one-line summary. The lines below are supposed to clarify a bit more, but lmk if you want me to reformat or something to make it more obvious

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37390043).

**Pull request closed without merging**